### PR TITLE
Adding Laser Tracker documentation

### PR DESCRIPTION
**Pull request by gmegh**

### CODE REVIEW

File: index.rst [hunk 1]
```diff
@@ -1,12 +1,102 @@
...
+
+To avoid these artifacts, we need to adopt a new method. 
+This method is detailed in one of the initial schmeatic drawings the vendor made, as seen below.
+Initially, we measure M1M3 and establish this measurement as the base or working 
+frame by setting the tracker at the center of M1M3. All subsequent measurements are 
+then made with respect to this M1M3 frame. Then, we proceed to measure M2 and the camera. 
+The measurement frame for M2 (`Frame_M2_meas`) is compared to the default `FrameM2`, 
+but crucially, this time the frame is centered at the actual location of the M2 target center. 
+This adjustment allows us to correct for any real displacements without the interference of 
+assumed distances between M1M3 and the target. It ensures that the default distance between
```
Comment (teweitsai) [inline index.rst:62]: The M2 can do the rigid body movement in the fine-tuning of optical system. In addition, its position would be changed at each time when performing the TMA rebalancing. Therefore, you might still get the artifact.

File: index.rst [hunk 2]
```diff
@@ -1,12 +1,145 @@
...
+around the Camera, M2, and M1M3.
+
+Operational Overview
+====================
+
+The operation of the Laser Tracker is facilitated through a Commandable SAL Component (CSC), 
+which interfaces with the custom T2SA software installed on the laser tracker computer. 
+The T2SA software on its turn interfaces with Spatial Analyzer, the proprietary software developed 
+by the laser tracker's manufacturer. Remote desktop access to the laser tracker computer is 
+available for operational control and monitoring, with access details provided within the
```
Comment (PaulinaLSST) [inline index.rst:26]: Could you provide the document's name which contains the operational control and monitoring?

File: index.rst [hunk 3]
```diff
@@ -1,12 +1,145 @@
...
+Operational Overview
+====================
+
+The operation of the Laser Tracker is facilitated through a Commandable SAL Component (CSC), 
+which interfaces with the custom T2SA software installed on the laser tracker computer. 
+The T2SA software on its turn interfaces with Spatial Analyzer, the proprietary software developed 
+by the laser tracker's manufacturer. Remote desktop access to the laser tracker computer is 
+available for operational control and monitoring, with access details provided within the 
+"lasetracker" Slack channel. This setup ensures a seamless operational 
+flow and allows for efficient management and troubleshooting of the system.
```
Comment (PaulinaLSST) [inline index.rst:28]: From drawing diagrams,  it could describe the system configuration and how the different components interact.? 
The information is clear, it's just a suggestion :)

File: index.rst [hunk 4]
```diff
@@ -1,12 +1,145 @@
...
+the center of the CAM target. 
+
+Indeed, take the example of 0.15deg rotation about x in the hexapod. This rotation would cause the
+FrameCAM at the center of M1M3 to move, 
+
+.. math::
+   
+   dY = \sin(0.15^\circ) \times \text{{distance between M1M3 and Camera}} \approx 0.15^\circ \times 4.06\,m \approx 10\,mm.
+
+In the following figure, we can observe the artifcats seen when moving the hexapod in dRY.
```
Comment (PaulinaLSST) [inline index.rst:49]: Does the camera sensitivity, have to be monitored during Observing Operation?

In the plot measured_offsets.png title there is a Typo: Sensitvity for Sensitivity

File: index.rst [hunk 5]
```diff
@@ -1,12 +1,145 @@
...
+At the Vera C. Rubin Observatory, the Laser Tracker plays a crucial role 
+in the precision alignment of the observatory's optical elements. 
+Positioned at the center of the M1M3 cell, it utilizes a laser to accurately 
+measure the positions of Spherically Mounted Retroreflectors (SMRs) distributed 
+around the Camera, M2, and M1M3.
+
+Operational Overview
+====================
+
+The operation of the Laser Tracker is facilitated through a Commandable SAL Component (CSC),
```
Comment (PaulinaLSST) [inline index.rst:22]: Could you please include an overview or image that illustrates the Laser Tracker, detailing its components? 
Or, could you provide the name of a document available in the Slack channel that discusses this?

File: index.rst [hunk 6]
```diff
@@ -1,12 +1,145 @@
...
+
+System Maintenance
+==================
+
+The laser tracker computer should not be powered down using the ``maintel/lasertracker/shut_down.py`` script. 
+Restarting the system requires physical presence at the observatory for manual reboot, a procedure designed 
+to minimize downtime and ensure continuous operational readiness.
+
+.. note:: For detailed instructions on accessing and operating the Laser Tracker Alignment System, 
+   please refer to the "lasetracker" Slack channel, where updates and access credentials are regularly provided.
```
Comment (PaulinaLSST) [inline index.rst:145]: Could you provide the document's name which contains instructions?

File: index.rst [hunk 7]
```diff
@@ -1,12 +1,145 @@
...
+  measured offsets until the specified tolerance is achieved.
+
+
+Special Procedures
+==================
+
+In addition to routine alignments, specific procedures are in place to ensure 
+comprehensive system readiness and operational integrity:
+
+- **BLOCK-246**: Tests that all targets are measurable by the laser tracker.
```
Comment (PaulinaLSST) [inline index.rst:128]: What should be done if Block-246 fails?

File: index.rst [hunk 8]
```diff
@@ -1,12 +1,145 @@
...
+==================
+
+In addition to routine alignments, specific procedures are in place to ensure 
+comprehensive system readiness and operational integrity:
+
+- **BLOCK-246**: Tests that all targets are measurable by the laser tracker.  
+  It goes through all the targets and ensures no errors are found when measuring them.
+
+
+- **BLOCK-197**: Tests the sequential misalignment and realignment of all optical elements,
```
Comment (PaulinaLSST) [inline index.rst:132]: What should be done if Block-197 fails?

File: index.rst [hunk 9]
```diff
@@ -1,12 +1,145 @@
...
+
+- **BLOCK-197**: Tests the sequential misalignment and realignment of all optical elements,   
+  testing and confirming the system's capability to accurately realign after intentional displacement.
+
+
+
+System Maintenance
+==================
+
+The laser tracker computer should not be powered down using the ``maintel/lasertracker/shut_down.py`` script.
```
Comment (PaulinaLSST) [inline index.rst:140]: What is the purpose of the ``maintel/lasertracker/shut_down.py`` script?

**Review by PaulinaLSST — COMMENTED** [review]
Hi Guillem 
Thanks for your document. 
It is an excellent document, my comments pertain to the view of OSs and daily commissioning duties, to well understand the information. 
I hope that you can review my comments and add them.

File: index.rst [hunk 10]
```diff
@@ -1,12 +1,145 @@
...
+Introduction
+============
 
-See the `Documenteer documentation <https://documenteer.lsst.io/technotes/index.html>`_ for tips on how to write and configure your new technote.
+At the Vera C. Rubin Observatory, the Laser Tracker plays a crucial role 
+in the precision alignment of the observatory's optical elements. 
+Positioned at the center of the M1M3 cell, it utilizes a laser to accurately 
+measure the positions of Spherically Mounted Retroreflectors (SMRs) distributed 
+around the Camera, M2, and M1M3.
+
```
Comment (JackieS-NL) [inline index.rst:18]: Do you think is a good idea to add that alignment of optical system is principally done by the wavefront sensor,  and the laser trackers is used to adjust if necessary, beyond what wavefront sensor can manage?

File: index.rst (hunk 5 above)
Comment (JackieS-NL) [reply index.rst:22]: agree. 

You could use image 15 of  https://docushare.lsstcorp.org/docushare/dsweb/Get/Document-21542/SPIE%209906-25%20M1M3%20Mirror%20Cell%20Assembly(2016)%20Submital%20Final.pdf

File: index.rst [hunk 11]
```diff
@@ -1,12 +1,145 @@
...
+Operational Overview
+====================
+
+The operation of the Laser Tracker is facilitated through a Commandable SAL Component (CSC), 
+which interfaces with the custom T2SA software installed on the laser tracker computer. 
+The T2SA software on its turn interfaces with Spatial Analyzer, the proprietary software developed
```
Comment (JackieS-NL) [inline index.rst:24]: “on its turn” could you rephrase this?

File: index.rst [hunk 12]
```diff
@@ -1,12 +1,145 @@
...
+============
 
-See the `Documenteer documentation <https://documenteer.lsst.io/technotes/index.html>`_ for tips on how to write and configure your new technote.
+At the Vera C. Rubin Observatory, the Laser Tracker plays a crucial role 
+in the precision alignment of the observatory's optical elements. 
+Positioned at the center of the M1M3 cell, it utilizes a laser to accurately 
+measure the positions of Spherically Mounted Retroreflectors (SMRs) distributed 
+around the Camera, M2, and M1M3.
+
+Operational Overview
```
Comment (JackieS-NL) [inline index.rst:19]: Great overview

File: index.rst (hunk 10 above)
Comment (JackieS-NL) [inline index.rst:18]: Would it be relevant to add a note that during commissioning, laser tracker is also used to measure the flexure of the telescope?

File: index.rst [hunk 13]
```diff
@@ -1,12 +1,145 @@
...
+available for operational control and monitoring, with access details provided within the 
+"lasetracker" Slack channel. This setup ensures a seamless operational 
+flow and allows for efficient management and troubleshooting of the system.  
+
+Computing Target offsets
+========================
+
+In our previous approach to computing offsets on M2 and the Camera, we employed a 
+method where we would measure the target (we will choose "Camera" for this explanation) points and update the reference `FrameCAM`
+by creating the `Frame_CAM_meas`. Notably, `FrameCAM` was centered at the origin of M1M3.
```
Comment (JackieS-NL) [inline index.rst:35]: * For me is hard to understand what are Frame[component], Frame_[component]_meas. Would you consider adding what they represent the first time you mention them?

File: index.rst [hunk 14]
```diff
@@ -1,12 +1,145 @@
...
+Indeed, take the example of 0.15deg rotation about x in the hexapod. This rotation would cause the
+FrameCAM at the center of M1M3 to move, 
+
+.. math::
+   
+   dY = \sin(0.15^\circ) \times \text{{distance between M1M3 and Camera}} \approx 0.15^\circ \times 4.06\,m \approx 10\,mm.
+
+In the following figure, we can observe the artifcats seen when moving the hexapod in dRY. 
+
+.. image:: /_static/measured_offsets.png
```
Comment (JackieS-NL) [inline index.rst:51]: Could you add a legend for the colored lines?

File: index.rst [hunk 15]
```diff
@@ -1,12 +1,145 @@
...
+   :align: center
+
+To avoid these artifacts, we need to adopt a new method. 
+This method is detailed in one of the initial schmeatic drawings the vendor made, as seen below.
```
Comment (JackieS-NL) [inline index.rst:55]: typo 'schmeatic'

File: index.rst [hunk 16]
```diff
@@ -1,12 +1,145 @@
...
+Initially, we measure M1M3 and establish this measurement as the base or working 
+frame by setting the tracker at the center of M1M3. All subsequent measurements are 
+then made with respect to this M1M3 frame. Then, we proceed to measure M2 and the camera.
```
Comment (JackieS-NL) [inline index.rst:58]: * “with respect ” —> relative?

File: index.rst (hunk 15 above)
Comment (JackieS-NL) [reply index.rst:55]: "seen below" - add a reference to link to the image?.

File: index.rst [hunk 17]
```diff
@@ -1,12 +1,145 @@
...
+
+.. image:: /_static/measured_offsets.png
+   :align: center
+
+To avoid these artifacts, we need to adopt a new method. 
+This method is detailed in one of the initial schmeatic drawings the vendor made, as seen below.
+Initially, we measure M1M3 and establish this measurement as the base or working 
+frame by setting the tracker at the center of M1M3. All subsequent measurements are 
+then made with respect to this M1M3 frame. Then, we proceed to measure M2 and the camera. 
+The measurement frame for M2 (`Frame_M2_meas`) is compared to the default `FrameM2`,
```
Comment (JackieS-NL) [inline index.rst:59]: What is the default FrameM2?

File: index.rst [hunk 18]
```diff
@@ -1,12 +1,145 @@
...
+Initially, we measure M1M3 and establish this measurement as the base or working 
+frame by setting the tracker at the center of M1M3. All subsequent measurements are 
+then made with respect to this M1M3 frame. Then, we proceed to measure M2 and the camera. 
+The measurement frame for M2 (`Frame_M2_meas`) is compared to the default `FrameM2`, 
+but crucially, this time the frame is centered at the actual location of the M2 target center. 
+This adjustment allows us to correct for any real displacements without the interference of 
+assumed distances between M1M3 and the target. It ensures that the default distance between 
+M1M3 and the target is not erroneously generating displacement artifacts.
+
+.. image:: /_static/drawing.png
```
Comment (JackieS-NL) [inline index.rst:65]: The diagram is very clear

File: index.rst [hunk 19]
```diff
@@ -1,12 +1,145 @@
...
+by the laser tracker's manufacturer. Remote desktop access to the laser tracker computer is 
+available for operational control and monitoring, with access details provided within the 
+"lasetracker" Slack channel. This setup ensures a seamless operational 
+flow and allows for efficient management and troubleshooting of the system.  
+
+Computing Target offsets
+========================
+
+In our previous approach to computing offsets on M2 and the Camera, we employed a 
+method where we would measure the target (we will choose "Camera" for this explanation) points and update the reference `FrameCAM`
```
Comment (JackieS-NL) [inline index.rst:34]: - “where we would” —> “involving” “to” maybe?
- (we will choose "Camera" for this explanation)”  could be in another place as it interrupts the phrase. 
- What differentiates the use of ‘Camera’ from another component in the explanation in this paragraph? * Would it be a possibility to give a generic name instead to describe this approach?

File: index.rst [hunk 20]
```diff
@@ -1,12 +1,145 @@
...
+   )
+
+Note that this also involves changing the default Z coordinate of `FrameCAM` and `FrameM2` to their nominal
+Z position with respect to M1M3.
+
+It is important to understand that the new method is doing two things:
+
+1. When measuring the target M1M3, it is setting the reference frame to the center of M1M3. 
+   This means any measurements after will be in that frame of reference. 
+2. We measure the offset between the default target offset, which is the optical optimal
```
Comment (JackieS-NL) [inline index.rst:103]: -  Would you consider placing the definition first, it MAY make it more clear. Just changing the order, like 
 “We measure the offset between the optical optimal position of the target w.r.t. M1M3 (this is the default target offset), and the measured target offset. 
-  w.r.t. to M1M3  --> w.r.t. M1M3

File: index.rst [hunk 21]
```diff
@@ -1,12 +1,145 @@
...
+to align all optical elements within the observatory. This operation is supported by a series 
+of specialized scripts designed to facilitate the alignment process:
+
+- ``maintel/lasertracker/measure.py``: This script measures a target  
+  and calculates the offset of that target (either M2 or Camera) with respect to M1M3.
+
+- ``maintel/lasertracker/align.py``: With inputs for a target and a tolerance level,  
+  this script measures the chosen target and iteratively realigns the telescope using the  
+  measured offsets until the specified tolerance is achieved.
+
```
Comment (JackieS-NL) [inline index.rst:120]: Is it possible that this operation is needed in the middle of observations?

File: index.rst (hunk 9 above)
Comment (JackieS-NL) [reply index.rst:140]: agree, 
Also maybe highlight ”should not be powered down”?

File: index.rst (hunk 6 above)
Comment (JackieS-NL) [inline index.rst:145]: Add link to slack channel.

**Review by JackieS-NL — CHANGES_REQUESTED** [review]
Hello Guillem 

The document is great, I like that explains the context along with the technical details.

I have many questions due to my unfamiliarity with this subject. **If any questions/suggestions are obvious, please ignore them**.

Also, I don’t know if this applies to tech notes: should images have captions?
